### DISCUSSION TITLE
Compare Blob Value in DAO

### DISCUSSION THREAD

**Discussion by yeanpei** [simolus3/drift#2478] [Q&A] [answered]
I have a column uuid which store as a blob in database, and convert to String using the UUIDConverter for entity.

`BlobColumn get uuid => blob().named("uuid").map(const UUIDConverter())();`

This is the UUID Converter class
```
class UUIDConverter extends TypeConverter<String, Uint8List> {
  const UUIDConverter();

  @override
  String fromSql(Uint8List fromDb) => Uuid.unparse(fromDb);

  @override
  Uint8List toSql(String value) =>
      Uint8List.fromList(Uuid.parse((value.isEmpty ? Uuid().v4() : value)));
}
```

In my DAO, when I want to use the column to do comparison, the type is still blob so I am not able to do String comparison.
I tried like below, but is there any other way to compare the value more efficiently?

```
Future<int> deleteByUuid(String uuid) {
    //delete row based on primary key
    return fieldInspIssues
        .deleteWhere((t) => t.uuid.equals(Uint8List.fromList(uuid.codeUnits)));
}
```

**Comment by simolus3** [comment, marked as answer]
To compare values on columns with a type converter, you can use the [`equalsValue`](https://drift.simonbinder.eu/api/drift/generatedcolumnwithtypeconverter/equalsvalue) method:

```dart
Future<int> deleteByUuid(String uuid) {
    //delete row based on primary key
    return fieldInspIssues
        .deleteWhere((t) => t.uuid.equalsValue(uuid.codeUnits));
}

```

**Reply by yeanpei** [reply]
I didn't notice got this method, thanks!
